graphql/cxs-impl/src/main/java/org/apache/unomi/graphql/condition/EventConditionFactory.java

```diff
@@ -17,93 +17,151 @@
             }
 
             if (filterInput.getOr() != null && filterInput.getOr().size() > 0) {
-                final Condition filterOrCondition = createBoolCondition("or", definitionsService);
-                final List<Condition> filterOrSubConditions = filterInput.getOr().stream()
-                        .map(orInput -> createEventFilterInputCondition(orInput, null, null, definitionsService))
-                        .collect(Collectors.toList());
-                filterOrCondition.setParameter("subConditions", filterOrSubConditions);
-                rootSubConditions.add(filterOrCondition);
+                rootSubConditions.add(filtersToCondition(filterInput.getOr(), orInput -> eventFilterInputCondition(orInput, null, null), "or"));
             }
         }
 
-        rootCondition.setParameter("subConditions", rootSubConditions);
-        return rootCondition;
+        return booleanCondition("and", rootSubConditions);
     }
 
-    private Condition createListUpdateEventCondition(CDPListsUpdateEventFilterInput cdp_listsUpdateEvent, DefinitionsService definitionsService) {
+    private Condition listUpdateEventCondition(CDPListsUpdateEventFilterInput cdp_listsUpdateEvent) {
 
         final List<Condition> rootSubConditions = new ArrayList<>();
 
         if (cdp_listsUpdateEvent.getJoinLists_contains() != null && !cdp_listsUpdateEvent.getJoinLists_contains().isEmpty()) {
-            rootSubConditions.add(createPropertiesCondition("joinLists", "contains", cdp_listsUpdateEvent.getJoinLists_contains(), definitionsService));
+            rootSubConditions.add(propertiesCondition("joinLists", "contains", cdp_listsUpdateEvent.getJoinLists_contains()));
         }
 
         if (cdp_listsUpdateEvent.getJoinLists_contains() != null && !cdp_listsUpdateEvent.getJoinLists_contains().isEmpty()) {
-            rootSubConditions.add(createPropertiesCondition("leaveLists", "contains", cdp_listsUpdateEvent.getLeaveLists_contains(), definitionsService));
+            rootSubConditions.add(propertiesCondition("leaveLists", "contains", cdp_listsUpdateEvent.getLeaveLists_contains()));
         }
 
-        final Condition rootCondition = createBoolCondition("and", definitionsService);
-        rootCondition.setParameter("subConditions", rootSubConditions);
-        return rootCondition;
+        return booleanCondition("and", rootSubConditions);
     }
 
+    private Condition createCdpConsentUpdateEventCondition(final CDPConsentUpdateEventFilterInput eventFilterInput) {
+        final GraphQLInputObjectType inputObjectType =
+                (GraphQLInputObjectType) environment.getGraphQLSchema().getType(CDPConsentUpdateEventFilterInput.TYPE_NAME);
+
+        final List<Condition> subConditions = new ArrayList<>();
+
+        if (eventFilterInput.getStatus_equals() != null) {
+            subConditions.add(propertyCondition("status", eventFilterInput.getStatus_equals()));
```

Comment:
here should be used `properties.status`. It's not your bug, but that should be fixed.